client.go

```diff
@@ -1375,3 +1398,455 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
 		RecordType:   recordType,
 	})
 }
+
+type SecretRequest struct {
+	SecretType  string
+	SecretName  string
+	SecretValue string
+	Description string
+	FileName    string
+}
+
+// CreateSecret makes a secret of the specified type and shares it with the client
+func (c *ToznySDKV3) CreateSecret(ctx context.Context, login TozIDLoginRequest, secret SecretRequest) (*pdsClient.Record, string, error) {
+	var createdSecret *pdsClient.Record
+	SECRET_TYPES := []string{"Client", "Credential", "File"}
+	// actually need to check the input for secret here too
+	secret.SecretName = strings.TrimSpace(secret.SecretName)
+	secret.SecretValue = strings.TrimSpace(secret.SecretValue)
+	if secret.SecretType == "" {
+		return createdSecret, "type cannot be empty", nil
+	}
+	if !Contains(SECRET_TYPES, secret.SecretType) {
+		return createdSecret, "invalid type", nil
+	}
+	if secret.SecretName == "" {
+		return createdSecret, "name cannot be empty", nil
+	}
+	matched, err := regexp.MatchString(`^[a-zA-Z0-9-_]{1,50}$`, secret.SecretName)
+	if err != nil {
+		return createdSecret, "", err
+	}
+	if !matched {
+		return createdSecret, "Secret name must contain 1-50 alphanumeric characters, -, or _", nil
+	}
+	if secret.SecretValue == "" && secret.SecretType == "Credential" {
+		return createdSecret, "Value cannot be empty", nil
+	}
+	if secret.SecretType == "File" && strings.TrimSpace(secret.SecretName) == "" {
+		return createdSecret, "File name cannot be empty", nil
+	}
+	// if (secret.SecretType == "File" && secret.File.size > 5242880) {
```

Comment:
this can be deleted right, since we don't have to worry about memory / file system limits as much on the server compared to in the browser?